lucene/core/src/java/org/apache/lucene/search/Weight.java

```diff
@@ -69,6 +69,21 @@ protected Weight(Query query) {
    */
   public abstract void extractTerms(Set<Term> terms);
 
+  /**
+   * Returns {@link Matches} for a specific document, or {@code null} if the document
+   * does not match the parent query
```

Comment:
maybe mention that a match without positions will be reported as an empty instance?

Reply:
++